app/controllers/planning_applications/review/recommendations_controller.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module PlanningApplications
+  module Review
+    class RecommendationsController < AuthenticationController
+      include CommitMatchable
+
+      before_action :set_planning_application
+      before_action :ensure_user_is_reviewer_checking_assessment, only: %i[update edit]
+      before_action :set_recommendations, only: %i[update edit]
+      before_action :set_committee_decision
+      before_action :set_recommendation, only: :update
+
+      def new
+        @recommendation = Recommendation.new
+      end
+
+      def edit
+        @recommendation = @recommendations.last
+
+        respond_to do |format|
+          format.html
+        end
+      end
+
+      def create
+        @recommendation = Recommendation.new(
+          planning_application: @planning_application.planning_application,
+          status: "review_complete",
+          reviewer: Current.user
+        )
+
+        if @recommendation.save
+          @planning_application.update!(decision: recommendation_form_params[:decision], public_comment: recommendation_form_params[:public_comment])
```

Comment:
This should be handled in the model layer - either detect the update needs to be made and use a callback or add an explicit method like this:

``` ruby
def save_and_submit(params)
  transaction do
    save!
    planning_application.update!(params.slice(:decision, :public_comment))
    planning_application.submit!
  end

  true
rescue ActiveRecord::RecordNotSaved
  false
end
```

Then here you can do:

``` ruby
if @recommendation.save_and_submit(recommendation_form_params)
  redirect_to planning_application_review_tasks_path(@planning_application), notice: t(".success")
else
  render :new
end
```

(the model method may need extra classes to rescue, etc.)